fabric-fluid-v1/src/main/java/net/fabricmc/fabric/api/fluid/v1/container/volume/FluidVolume.java

```diff
@@ -0,0 +1,42 @@
+package net.fabricmc.fabric.api.fluid.v1.container.volume;
+
+import java.util.Iterator;
+
+import com.google.common.collect.Iterators;
+
+import net.minecraft.fluid.Fluid;
+import net.minecraft.nbt.CompoundTag;
+
+import net.fabricmc.fabric.api.fluid.v1.container.FluidContainer;
+
+/**
+ * A fluid container that can only hold one fluid.
+ */
+public interface FluidVolume extends FluidContainer {
+	@Override
+	default long getTotalVolume() {
+		return this.getAmount();
+	}
+
+	/**
+	 * @return the amount of fluid currently in the fluid volume.
+	 */
+	long getAmount();
+
+	@Override
+	default Iterator<FluidVolume> iterator() {
+		return Iterators.singletonIterator(this);
+	}
+
+	@Override
+	default FluidVolume simpleCopy() {
+		return new SimpleFluidVolume(this.getFluid(), this.getAmount(), this.getData() == null ? null : this.getData().copy());
+	}
+
+	Fluid getFluid();
+
+	/**
+	 * @return the data associated with the fluid, may be null
+	 */
+	CompoundTag getData();
```

Comment:
We should really have `getProperty` and `setProperty` methods instead, I think. That'd make it a *lot* harder to make bad data that causes crashes on deserialization, and make the jobs of mods which want to track property changes a *lot* easier.

Reply:
good idea